### PR TITLE
OIDC federated IdP initiated backchannel logout 

### PR DESCRIPTION
**Pull request by ThaminduR**
### Proposed changes in this pull request

Implement the OIDC federated identity provider initiated logout feature.

Related Issue - https://github.com/wso2/product-is/issues/9254

### CODE REVIEW

**Review by piraveena — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 1]
```diff
@@ -224,6 +224,11 @@
                 <artifactId>nimbus-jose-jwt</artifactId>
                 <version>${nimbusds.version}</version>
             </dependency>
+            <dependency>
+                <groupId>com.nimbusds</groupId>
+                <artifactId>oauth2-oidc-sdk</artifactId>
+                <version>8.28.3</version>
```
Comment (piraveena) [inline pom.xml:230]: move version separated and use a placeholder

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+
+    protected X509Certificate resolveSignerCertificate(JWSHeader header,
+                                                       IdentityProvider idp) throws LogoutServerException {
+
+        X509Certificate x509Certificate = null;
+        try {
+            x509Certificate = (X509Certificate) IdentityApplicationManagementUtil
+                    .decodeCertificate(idp.getCertificate());
+        } catch (CertificateException e) {
+
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java:543]: remove new line

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Signature Algorithm not supported yet : " + alg);
+                    }
+                }
+                if (verifier == null) {
+                    throw new LogoutServerException("Could not create a signature verifier for algorithm type: " + alg);
+                }
+            }
+
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java:498]: remove new line

**Review by piraveena — CHANGES_REQUESTED** [review]

File: components/org.wso2.carbon.identity.application.authenticator.oidc/pom.xml [hunk 4]
```diff
@@ -156,7 +162,13 @@
                             version="${identity.framework.package.import.version.range}",
                             org.wso2.carbon.ui; version="${carbon.kernel.package.import.version.range}",
                             com.nimbusds.jose.*; version="${nimbusds.osgi.version.range}",
-                            com.nimbusds.jwt; version="${nimbusds.osgi.version.range}"
+                            com.nimbusds.jwt; version="${nimbusds.osgi.version.range}",
+                            com.nimbusds.oauth2.sdk.id.*;
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/pom.xml:166]: Any reason why we haven't added the OSGi version range?

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/LogoutClientException.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.oidc;
+
+import org.wso2.carbon.identity.application.authentication.framework.exception.FrameworkException;
+
+public class LogoutClientException extends FrameworkException {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/LogoutClientException.java:23]: Class level comment

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java [hunk 6]
```diff
@@ -54,6 +56,9 @@ private OIDCAuthenticatorConstants() {
 
     public static final String SID = "sid";
 
+    public static final Pattern OIDC_BCLOGOUT_ENDPOINT_URL_PATTERN =
+            Pattern.compile("(.*)/identity/oidc/slo?");
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java:60]:
```suggestion
            Pattern.compile("(.*)/identity/oidc/slo(.*)");
```

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java [hunk 7]
```diff
@@ -104,4 +109,14 @@ private Claim() {
...
+    public class JWT {
+
+        private JWT() {
+
+        }
+
+        public static final String JWKS_VALIDATION_ENABLE_CONFIG = "JWTValidatorConfigs.Enable";
+        public static final String JWKS_URI = "jwksUri";
+    }
 }
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java:122]:
```suggestion
}

```

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    private static final Log log = LogFactory.getLog(SessionInfoDAO.class);
+
+    /**
+     * Retrieve the session details of a given OIDC sid from the database.
+     *
+     * @param oidcSId Session Id of the OIDC Logout Request.
+     * @return Map of session details.
+     * @throws LogoutServerException If DB execution fails.
+     */
+    public Map<String, String> getSessionDetails(String oidcSId) throws LogoutServerException {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java:47]: This DAO seems to required in two places as of now. What about moving this to the framework?
@piraveena

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java (hunk 6 above)
Comment (piraveena) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java:60]: need to have a framework PR for this endpoint as well. please link that

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutRequestFactory.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+public class LogoutRequestFactory extends HttpIdentityRequestFactory {
+
+    private static final Log log = LogFactory.getLog(LogoutRequestFactory.class);
+
+    @Override
+    public boolean canHandle(HttpServletRequest request, HttpServletResponse response) {
+
+        boolean canHandle = false;
+        if (request != null) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutRequestFactory.java:43]: I don't this we need this null check

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java (hunk 8 above)
Comment (piraveena) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java:47]: yes we can move this to framework and refactor saml as well

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+    private static final Log log = LogFactory.getLog(LogoutResponseFactory.class);
+
+    @Override
+    public boolean canHandle(IdentityResponse identityResponse) {
+
+        if (identityResponse instanceof LogoutResponse) {
+            return true;
+        }
+        return false;
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java:46]:
```suggestion
        return (identityResponse instanceof LogoutResponse) 
```

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java (hunk 8 above)
Comment (piraveena) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java:47]: can we rename ths oidcSId ->fedIdpSessionId as well

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    public boolean canHandle(FrameworkException exception) {
+
+        if (exception instanceof LogoutServerException || exception instanceof LogoutClientException) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java:51]: We can have a parent class for LogoutServerException and LogoutClientException and check instance of that here.

File: pom.xml [hunk 12]
```diff
@@ -224,6 +224,11 @@
                 <artifactId>nimbus-jose-jwt</artifactId>
                 <version>${nimbusds.version}</version>
             </dependency>
+            <dependency>
```
Comment (mefarazath) [inline pom.xml:227]: Is this dependency available in the runtime?

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java [hunk 13]
```diff
@@ -0,0 +1,549 @@
...
+import java.util.regex.Matcher;
+
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.AnalyticsAttributes.SESSION_ID;
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.AUTHENTICATOR_NAME;
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.OIDC_BCLOGOUT_ENDPOINT_URL_PATTERN;
+
+public class LogoutProcessor extends IdentityProcessor {
+
+    private static final Log log = LogFactory.getLog(LogoutProcessor.class);
+    private SessionManagementService sessionManagementService = new SessionManagementService();
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java:79]: We should get this service via OSGi declarative service.

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java [hunk 14]
```diff
@@ -0,0 +1,549 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.AnalyticsAttributes.SESSION_ID;
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.AUTHENTICATOR_NAME;
+import static org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.OIDC_BCLOGOUT_ENDPOINT_URL_PATTERN;
+
+public class LogoutProcessor extends IdentityProcessor {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java:76]: Does the name convey the purpose?
LogoutProcessor seems too generic to me. FederatedIdpInitLogoutProcessor or something along the lines would give the idea that we are dealing with federated idp logout request IMO.

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java [hunk 15]
```diff
@@ -0,0 +1,549 @@
...
+    public IdentityResponse.IdentityResponseBuilder process(IdentityRequest identityRequest) throws FrameworkException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Request processing started by OIDC LogoutProcessor.");
+        }
+        LogoutContext logoutContext = new LogoutContext(identityRequest);
+
+        IdentityResponse.IdentityResponseBuilder identityResponseBuilder = null;
+
+        if (identityRequest instanceof LogoutRequest) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java:104]: Do we need this check?
We can only reach here if this was a logout request, isn't it? (Processors canHandle() method should do this)

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java [hunk 16]
```diff
@@ -0,0 +1,549 @@
...
+                log.debug("Can't handle the request");
+            }
+            throw new LogoutClientException("Can't handle the request");
+        }
+
+        return identityResponseBuilder;
+
+    }
+
+    protected IdentityResponse.IdentityResponseBuilder oidcFederatedLogout(
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java:117]: We can improve the method name IMO.

handleOIDCFederatedIdpInitLogout or handleOIDCFederatedLogoutRequest or handleOIDCFederatedLogout would be better names?

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java [hunk 17]
```diff
@@ -0,0 +1,549 @@
...
+            LogoutContext logoutContext) throws LogoutClientException, LogoutServerException {
+
+        LogoutResponse.LogoutResponseBuilder logoutResponseBuilder =
+                new LogoutResponse.LogoutResponseBuilder();
+        LogoutRequest logoutRequest = (LogoutRequest) logoutContext.getIdentityRequest();
+        try {
+            String logoutToken = logoutRequest.getParameter("logout_token");
+            if (StringUtils.isNotBlank(logoutToken)) {
+                log.info("Logout Token: " + logoutToken);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java:126]: This should be a DEBUG log

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java (hunk 16 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java]: fixed in 201c3de

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java (hunk 15 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java]: fixed in 201c3de

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java (hunk 14 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java]: fixed in c6b3fc7

File: pom.xml (hunk 12 above)
Comment (ThaminduR) [reply pom.xml:227]: dependecy is removed in 789a83b

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java (hunk 10 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java]: fixed in a8d1c44

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutRequestFactory.java (hunk 9 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutRequestFactory.java]: fixed in a8d1c44

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java (hunk 6 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java]: fixed in a8d1c44

File: components/org.wso2.carbon.identity.application.authenticator.oidc/pom.xml (hunk 4 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/pom.xml]: dependecy is removed in 789a83b

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java (hunk 3 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java]: fixed in 789a83b

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java (hunk 2 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java]: fixed in 789a83b

File: pom.xml (hunk 1 above)
Comment (ThaminduR) [reply pom.xml]: fixed in 789a83b

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+        }
+    }
+
+    public HttpIdentityResponse.HttpIdentityResponseBuilder handleException(FrameworkException frameworkException) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder builder =
+                new HttpIdentityResponse.HttpIdentityResponseBuilder();
+
+        if (frameworkException instanceof LogoutServerException) {
+            builder.setBody("Back channel logout failed due to server error.");
```
[the remark this answers is not in the document]
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java]: a96d285, a8d1c44

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+
+        final String query = "SELECT * FROM IDN_FED_AUTH_SESSION_MAPPING WHERE IDP_SESSION_ID = ?";
```
[the remark this answers is not in the document]
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java]: fixed in 789a83b

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java (hunk 7 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java]: b415003

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java (hunk 11 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java]: a6a859e

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java (hunk 13 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java]: `SessionManagementService` is not available as an OSGi service .

https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/SessionManagementService.java

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java (hunk 8 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java:48]: Moved the function to UserSessionDAO https://github.com/wso2/carbon-identity-framework/pull/3314

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java (hunk 6 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OIDCAuthenticatorConstants.java]: PR for the endpoint - https://github.com/wso2/carbon-identity-framework/pull/3322

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/TestUtils.java [hunk 20]
```diff
@@ -0,0 +1,17 @@
+package org.wso2.carbon.identity.application.authenticator.oidc;
+
```
[the remark this answers is not in the document]
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/TestUtils.java:20]: fixed in 71d21e4

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java (hunk 13 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/processor/LogoutProcessor.java]: https://github.com/wso2/carbon-identity-framework/pull/3321

**Pull request closed without merging**